receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,337 @@
+	if c.logs != nil {
+		if err := c.logs.Start(ctx, host); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+// Shuts Down the Alerts and Log combined receiver
+func (c *combindedLogsReceiver) Shutdown(ctx context.Context) error {
+	var errs error
+
+	// If we have an alerts receiver shutdown alerts collection
+	if c.alerts != nil {
+		if err := c.alerts.Shutdown(ctx); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	// If we have a logging receiver shutdown log collection
+	if c.logs != nil {
+		if err := c.logs.Shutdown(ctx); err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	return errs
+}
+
+func (s *receiver) createProjectMap() (map[string]*Project, error) {
+	// create a map of Projects
+	projects := make(map[string]*Project)
+
+	for _, project := range s.cfg.Logs.Projects {
+		projects[project.Name] = project
+	}
+
+	return projects, nil
+}
+
+// function parses out the hostname from the specified cluster host
+func FilterHostName(s string) []string {
+	var hostnames []string
+
+	// first check to make sure string is of adequate size
+	if len(s) > 0 {
+		// create an array with a comma delimiter from the original string
+		tmp := strings.Split(s, ",")
+		for _, t := range tmp {
+
+			u, err := url.Parse(t)
+			if err != nil {
+				fmt.Printf("Error parsing out %s", t)
+			}
+
+			// separate hostname from scheme and port
+			host, _, err := net.SplitHostPort(u.Host)
+			if err != nil {
+				// the scheme prefix was not added on to this string
+				// thus the hostname will have been placed under the "Scheme" field
+				hostnames = append(hostnames, u.Scheme)
+			} else {
+				// hostname parsed successfully
+				hostnames = append(hostnames, host)
+			}
+
+		}
+	}
+
+	return hostnames
+}
+
+// Actual functionality the receiver executes from the Start function
+func (s *receiver) KickoffReceiver(ctx context.Context) {
+	stopper := make(chan struct{})
+	s.stopperChanList = append(s.stopperChanList, stopper)
+	cfgProjects, err := s.createProjectMap()
+	if err != nil {
+		s.log.Error("Failed to create project map", zap.Error(err))
+	}
+
+	for {
+		// reterive all organizations listed in the MongoDB client
+		orgs, err := s.client.Organizations(ctx)
+		if err != nil {
+			s.log.Error("Error retrieving organizations", zap.Error(err))
+		}
+		// get all projects listed for each of the organizations
+		for _, org := range orgs {
+			projects, err := s.client.Projects(ctx, org.ID)
+			if err != nil {
+				s.log.Error("Error retrieving projects", zap.Error(err))
+			}
+
+			// filter out any projects not specified in the config
+			filteredProjects := make(map[string]*mongodbatlas.Project)
+			for _, project := range projects {
+				if _, ok := cfgProjects[project.Name]; ok {
+					filteredProjects[project.Name] = project
+				}
+			}
+
+			// get clusters for each of the projects
+			for _, project := range filteredProjects {
+				resource := resourceInfo{Org: org, Project: project}
```

Comment:
nit and not necessarily blocking: this method is pretty large. could we break out the stuff in this for loop into its own private function or something?